Code/Framework/AzFramework/AzFramework/TargetManagement/TargetManagementComponent.cpp

```diff
@@ -444,31 +349,144 @@ namespace AzFramework
+            return true;
+        }
+
+        // If we're a client, set the host to our desired target
+        if (connection->GetConnectionRole() == AzNetworking::ConnectionRole::Connector)
+        {
+            if (GetTargetInfo(packet.GetPersistentId()).GetPersistentId() == 0)
+            {
+                TargetContainer::pair_iter_bool ret = m_availableTargets.insert_key(packet.GetPersistentId());
+
+                TargetInfo& ti = ret.first->second;
+                ti.m_lastSeen = time(nullptr);
+                ti.m_displayName = "Host";
+                ti.m_persistentId = packet.GetPersistentId();
+                ti.m_networkId = static_cast<uint32_t>(connection->GetConnectionId());
+                ti.m_flags |= TF_ONLINE | TF_DEBUGGABLE | TF_RUNNABLE;
+
+                // last target can be loaded in with a persistent id but no network, correct here
+                if (GetDesiredTarget().GetPersistentId() == packet.GetPersistentId())
+                {
+                    m_settings->m_lastTarget.m_networkId = ti.m_networkId;
+                }
+
+                EBUS_QUEUE_EVENT(TargetManagerClient::Bus, TargetJoinedNetwork, ti);
+            }
+
+            if (GetDesiredTarget().GetPersistentId() != packet.GetPersistentId())
+            {
+                SetDesiredTarget(packet.GetPersistentId());
+            }
+        }
+
+        const uint32_t totalBufferSize = packet.GetSize();
+
+        // Messages can be larger than the size of a packet so reserve a buffer for the total message size
+        if (m_tmpInboundBufferPos == 0)
+        {
+            m_tmpInboundBuffer.reserve(totalBufferSize);
+        }
+
+        // Read as much data as the packet can include and append it to the buffer
+        const uint32_t readSize = AZStd::min(totalBufferSize - m_tmpInboundBufferPos, Neighborhood::NeighborBufferSize);
+        memcpy(m_tmpInboundBuffer.begin() + m_tmpInboundBufferPos, packet.GetMessageBuffer().GetBuffer(), readSize);
+        m_tmpInboundBufferPos += readSize;
+        if (m_tmpInboundBufferPos == totalBufferSize)
+        {
+            // Deserialize the complete buffer
+            AZ::IO::MemoryStream msgBuffer(m_tmpInboundBuffer.data(), totalBufferSize, totalBufferSize);
+            TmMsg* msg = nullptr;
+            AZ::ObjectStream::ClassReadyCB readyCB(AZStd::bind(&TargetManagementComponent::OnMsgParsed, this, &msg, AZStd::placeholders::_1,
+                AZStd::placeholders::_2, AZStd::placeholders::_3));
+            AZ::ObjectStream::LoadBlocking(&msgBuffer, *m_serializeContext, readyCB,
+                AZ::ObjectStream::FilterDescriptor(nullptr, AZ::ObjectStream::FILTERFLAG_IGNORE_UNKNOWN_CLASSES));
+
+            // Append to the inbox for handling
+            if (msg)
+            {
+                if (msg->GetCustomBlobSize() > 0)
+                {
+                    void* blob = azmalloc(msg->GetCustomBlobSize(), 1, AZ::OSAllocator, "TmMsgBlob");
```

Comment:
Who has responsibility for deleting this? Does the TmMsg own cleanup?

Reply:
Correct, on destruction.

Reply:
Where is the code that performs the destruction located at?
Is it using `azfree`?

Reply:
Correct, it's in TmMsg's dtor. This bit of logic was carried over from the previous implementation so the diff is a little non-obvious.